src/test/groovy/org/gradle/android/RoomSchemaLocationKspWorkaroundTest.groovy

```diff
@@ -0,0 +1,207 @@
+        Assume.assumeFalse(androidVersion >= VersionNumber.parse("7.2.0-alpha01") && kotlinVersionNumber < VersionNumber.parse("1.5.0"))
+        // Using Kotlin 1.8.0 with Ksp 1.7.22 throws java.lang.NoClassDefFoundError: org/jetbrains/kotlin/gradle/dsl/KotlinJvmOptionsImpl
+        Assume.assumeFalse(kotlinVersionNumber >= VersionNumber.parse("1.8.0") && kspVersionNumber == VersionNumber.parse("1.7.22-1.0.8"))
+
+        SimpleAndroidApp.builder(temporaryFolder.root, cacheDir)
+            .withAndroidVersion(androidVersion)
+            .withKotlinVersion(VersionNumber.parse(kotlinVersion))
+            .withKspVersion(kspVersion)
+            .build()
+            .writeProject()
+
+        cacheDir.deleteDir()
+        cacheDir.mkdirs()
+
+        when:
+        BuildResult buildResult = withGradleVersion(TestVersions.latestSupportedGradleVersionFor(androidVersion).version)
+            .forwardOutput()
+            .withProjectDir(temporaryFolder.root)
+            .withArguments(CLEAN_BUILD)
+            .build()
+
+        then:
+        assertCompileTasksHaveOutcome(buildResult, SUCCESS)
+        assertCompileAndroidTestTasksHaveOutcome(buildResult, SUCCESS)
+        assertCompileUnitTestTasksHaveOutcome(buildResult, SUCCESS)
+        assertKspTasksHaveOutcome(buildResult, SUCCESS)
+        assertKspAndroidTestTasksHaveOutcome(buildResult, SUCCESS)
+        assertKspUnitTestTasksHaveOutcome(buildResult, SUCCESS)
+        buildResult.task(':app:mergeRoomSchemaLocations').outcome == SUCCESS
+        buildResult.task(':library:mergeRoomSchemaLocations').outcome == SUCCESS
+
+        and:
+        assertKspSchemaOutputsExist()
+
+        and:
+        assertMergedSchemaOutputsExist()
+
+        when:
+        buildResult = withGradleVersion(TestVersions.latestSupportedGradleVersionFor(androidVersion).version)
+            .forwardOutput()
+            .withProjectDir(temporaryFolder.root)
+            .withArguments(CLEAN_BUILD)
+            .build()
+
+        then:
+        assertCompileTasksHaveOutcome(buildResult, FROM_CACHE)
+        assertCompileAndroidTestTasksHaveOutcome(buildResult, FROM_CACHE)
+        assertCompileUnitTestTasksHaveOutcome(buildResult, FROM_CACHE)
+        assertKspTasksHaveOutcome(buildResult, FROM_CACHE)
+        assertKspAndroidTestTasksHaveOutcome(buildResult, FROM_CACHE)
+        assertKspUnitTestTasksHaveOutcome(buildResult, FROM_CACHE)
+        buildResult.task(':app:mergeRoomSchemaLocations').outcome == SUCCESS
+        buildResult.task(':library:mergeRoomSchemaLocations').outcome == SUCCESS
+
+        and:
+        assertKspSchemaOutputsExist()
+
+        and:
+        assertMergedSchemaOutputsExist()
+
+        where:
+        //noinspection GroovyAssignabilityCheck
+        [androidVersion, kotlinVersion, kspVersion] << [TestVersions.latestAndroidVersions, TestVersions.supportedKotlinVersions, TestVersions.supportedKspVersions].combinations()
```

Comment:
maybe we can use a pair to group the `kotlinVersion` with its corresponding `kspVersion` so that KSP and Kotlin versions are tested with compatible versions? This will also reduce the explosion of the matrix here.

Reply:
good point,  
my initial approach was to test the combinations of Kotlin and Ksp, the results were:
|               | Ksp 1.7.22-1.0.8 | Ksp 1.8.10-1.0.9 |
|---------------|------------------|------------------|
| Kotlin 1.6.21 | OK               | OK               |
| Kotlin 1.7.21 | OK               | OK               |
| Kotlin 1.8    | KO               | OK               |

I guess I was a bit greedy and actually, some builds were failing with the daemon [exhausted](https://ge.solutions-team.gradle.com/s/6pn5ttx6hkrqy/tests/:testAndroid7_3_1/org.gradle.android.RoomSchemaLocationWorkaroundTest/schemas%20are%20generated%20into%20task-specific%20directory%20and%20are%20cacheable%20with%20kotlin%20and%20kapt%20workers%20disabled%20(Android%207.3.1)%20(Kotlin%201.8.0)?focused-exception-line=0-1-2-0&top-execution=1)

following https://github.com/gradle/android-cache-fix-gradle-plugin/pull/429/commits/250eb1a2f828caf68e6a26894b286e1d0be49053, we are testing only the pairs (`["1.6.21": "", "1.7.21": "1.7.22-1.0.8", "1.8.0": "1.8.10-1.0.9"]`)